dev/AppNotifications/AppNotificationManager.cpp

```diff
@@ -0,0 +1,358 @@
+#include <winrt/Windows.Storage.h>
+#include <winrt/base.h>
+#include "externs.h"
+#include "PushNotificationUtility.h"
+#include "AppNotificationUtility.h"
+#include <frameworkudk/pushnotifications.h>
+#include <frameworkudk/toastnotifications.h>
+#include <FrameworkUdk/toastnotificationsrt.h>
+#include "NotificationProperties.h"
+#include "NotificationTransientProperties.h"
+#include "AppNotification.h"
+#include <NotificationProgressData.h>
+#include <algorithm>
+#include <winerror.h>
+#include <string_view>
+#include <winrt/Windows.Foundation.Collections.h>
+
+using namespace std::literals;
+
+constexpr std::wstring_view expectedAppServerArgs = L"----AppNotificationActivated:"sv;
+
+namespace winrt
+{
+    using namespace winrt::Windows::UI;
+    using namespace winrt::Windows::Foundation;
+    using namespace winrt::Windows::Foundation::Collections;
+    using namespace winrt::Microsoft::Windows::AppNotifications;
+	using namespace Windows::ApplicationModel::Core;
+}
+
+namespace ToastABI
+{
+    using namespace ::ABI::Microsoft::Internal::ToastNotifications;
+    using namespace ::ABI::Windows::Foundation::Collections;
+}
+
+namespace PushNotificationHelpers
+{
+    using namespace winrt::Microsoft::Windows::PushNotifications::Helpers;
+}
+
+using namespace Microsoft::Windows::AppNotifications::Helpers;
+
+namespace winrt::Microsoft::Windows::AppNotifications::implementation
+{
+    winrt::Microsoft::Windows::AppNotifications::AppNotificationManager AppNotificationManager::Default()
+    {
+        static auto appNotificationManager{winrt::make<AppNotificationManager>()};
+        return appNotificationManager;
+    }
+
+    void AppNotificationManager::Register()
+    {
+        std::wstring storedComActivatorString;
+        if (!PushNotificationHelpers::IsPackagedAppScenario())
+        {
+            std::wstring notificationId{ RetrieveNotificationAppId() };
+            if (!AppModel::Identity::IsPackagedProcess())
+            {
+                THROW_IF_FAILED(PushNotifications_RegisterFullTrustApplication(notificationId.c_str(), GUID_NULL));
+
+                storedComActivatorString = RegisterComActivatorGuidAndAssets();
+                // Remove braces around the guid string
+                storedComActivatorString = storedComActivatorString.substr(1, storedComActivatorString.size() - 2);
+            }
+
+            wil::unique_cotaskmem_string processName;
+            THROW_IF_FAILED(GetCurrentProcessPath(processName));
+            auto notificationPlatform{ PushNotificationHelpers::GetNotificationPlatform() };
+            THROW_IF_FAILED(notificationPlatform->AddToastRegistrationMapping(processName.get(), notificationId.c_str()));
+        }
+
+        winrt::guid registeredClsid{ GUID_NULL };
+        if (AppModel::Identity::IsPackagedProcess())
+        {
+            THROW_IF_FAILED(PushNotificationHelpers::GetComRegistrationFromRegistry(expectedAppServerArgs.data(), registeredClsid));
```

Comment:
>GetComRegistrationFromRegistry [](http://example.com/codeflow?start=53&length=30)

This method could return the `winrt::guid` rather than emitting it as an out-parameter.